controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -89,6 +108,50 @@ func (f *admissionControllerFeature) Configure(dda *v2alpha1.DatadogAgent) (reqC
 		if ac.WebhookName != nil {
 			f.webhookName = *ac.WebhookName
 		}
+
+		sidecarConfig := dda.Spec.Features.AdmissionController.AgentSidecarInjection
+		if shouldEnablesidecarInjection(sidecarConfig) {
+			f.agentSidecarConfig = &AgentSidecarInjectionConfig{}
+			if sidecarConfig.Enabled != nil {
+				f.agentSidecarConfig.enabled = *sidecarConfig.Enabled
+
+			}
+			if sidecarConfig.Provider != nil && *sidecarConfig.Provider != "" {
+				f.agentSidecarConfig.provider = *sidecarConfig.Provider
+			}
+
+			if sidecarConfig.ClusterAgentCommunicationEnabled != nil {
+				f.agentSidecarConfig.clusterAgentCommunicationEnabled = *sidecarConfig.ClusterAgentCommunicationEnabled
+			}
+			// set image registry from admissionController config or global config if defined
+			if sidecarConfig.Registry != nil && *sidecarConfig.Registry != "" {
+				f.agentSidecarConfig.registry = *sidecarConfig.Registry
+			} else if dda.Spec.Global.Registry != nil && *dda.Spec.Global.Registry != "" {
+				f.agentSidecarConfig.registry = *dda.Spec.Global.Registry
+			}
+
+			// set agent image from admissionController config or nodeAgent override image name. else, It will follow agent image name.
+			// default is "agent"
+			componentOverride, ok := dda.Spec.Override[v2alpha1.NodeAgentComponentName]
+			if sidecarConfig != nil && sidecarConfig.Image != nil && sidecarConfig.Image.Name != "" {
```

Comment:
nit - this and next block can be merged since they both depend on config and image not being nil
`sidecarConfig != nil && sidecarConfig.Image != nil`.